sdk/servicebus/azure-servicebus/azure/servicebus/_common/receiver_mixins.py

```diff
@@ -46,10 +46,7 @@ def _populate_attributes(self, **kwargs):
 
         self._auth_uri = "sb://{}/{}".format(self.fully_qualified_namespace, self.entity_path)
         self._entity_uri = "amqps://{}/{}".format(self.fully_qualified_namespace, self.entity_path)
-        self._receive_mode = kwargs.get("receive_mode", ReceiveMode.PeekLock)
-        # While we try to leave failures to the service, in this case the errors lower down the stack are less clear.
-        if not isinstance(self._receive_mode, ReceiveMode):
-            raise TypeError("Parameter 'receive_mode' must be of type ReceiveMode")
+        self._receive_mode = ServiceBusReceiveMode(kwargs.get("receive_mode", ServiceBusReceiveMode.PeekLock))
```

Comment:
case sensitivity when user passes string?
e.g. if user passes "PeekLock" instead "peeklock", I assume there would be a ValueError?
do we want to convert the string to lower case first?

Reply:
It will ValueError, correct.  I'm not apt to convert because eventually if we onboard to "Fancy Future Enums" that'll come for us, and would be additive?  If you agree that it would be non-breaking I'm inclined to let that come via autorest if only because it being... basically code-complete day :P

Reply:
(Edit, I literally just re-read the discussion with Anna I had above and she seems already on board with this plan, so LMK if you concur)

Reply:
yep, I see the conversation, sounds good to me